### PR TITLE
first version

### PR DESCRIPTION
**Pull request by jgfidelis**
https://badge--storecomponents.myvtex.com/work-shirt/p

### CODE REVIEW

File: react/components/BaseSpecificationBadges.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+import slugify from '../modules/slug'
+
+import styles from './styles.css'
+
+interface Props {
+  product?: Product
+}
+
+const checkConditionForSpecification = (condition: Condition, specification: Specification) => {
+  const { type, value, show } = condition
```
Comment (iaronaraujo) [inline react/components/BaseSpecificationBadges.tsx]: Could `type` be an enum?

File: react/components/BaseSpecificationBadges.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  blockClass,
+  orientation = 'vertical'
+}) => {
+  const badges = getVisibleBadges(product, conditions, groupName)
+
+  if (!product || !conditions || badges.length === 0) {
+    return null
+  }
+
+  const isVertical = orientation === 'vertical'
```
Comment (iaronaraujo) [inline react/components/BaseSpecificationBadges.tsx]: Could be this an enum too?

Comment (iaronaraujo) [reply react/components/BaseSpecificationBadges.tsx]: If there are only 2 cases maybe a boolean?

**Review by iaronaraujo — APPROVED** [review]

File: manifest.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{
+  "vendor": "vtex",
+  "name": "product-specification-badges",
+  "version": "0.0.1",
+  "title": "Product Specification Badges",
+  "description": "Display Product Specification Badges",
+  "defaultLocale": "pt-BR",
```
Comment (victorhmp) [inline manifest.json:7]: Not sure about what this does hahaha but if it has to be here, `en-US` would be a better choice

File: react/components/BaseSpecificationBadges.tsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+      if (checkConditionForSpecification(condition, spec)) {
+        return { specification: spec, show: condition.show }
+      }
+      return null
+    }).filter(Boolean) as VisibleSpecification[]
+  }
+
+  const conditionsArray = conditions as Condition[]
+  return conditionsArray.map(cond =>
+    getValidSpecificationForCondition(cond, group!.specifications))
```
Comment (victorhmp) [inline react/components/BaseSpecificationBadges.tsx]: This function call could return `null`, wouldn't `.filter` result in a runtime error?

**Review by victorhmp — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -1,3 +1,247 @@
 # VTEX Product Specification Badges
 
-Use this component to show badges based on your product specifications.
+Use this component to show badges based on your product specifications inside your product page.
```
Comment (lbebber) [inline README.md]: You can remove this file, Github is smart and displays the one from `docs/README` if needed

File: react/components/BaseSpecificationBadges.tsx (hunk 2 above)
Comment (lbebber) [reply react/components/BaseSpecificationBadges.tsx]: @iaronaraujo even though it makes sense to use boolean here, I think it's better to use string as a value, makes for a nicer API (compare `isVertical: false` vs `orientation: horizontal`)

File: react/components/BaseSpecificationBadges.tsx [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+        if (displayValue === 'SPECIFICATION_VALUE') {
+          valueToShow = badge.specification.values[0]
+        }
+
+        if (displayValue === 'SPECIFICATION_NAME') {
+          valueToShow = badge.specification.name
+        }
+
+        if (!displayValue) {
+          return null
```
Comment (lbebber) [inline react/components/BaseSpecificationBadges.tsx:148]: perhaps add a warning on the console here

Comment (lbebber) [inline react/components/BaseSpecificationBadges.tsx:148]:
```suggestion
          console.warn('You need to set a `displayValue` for the `product-specification-badges` block, either `SPECIFICATION_VALUE` or `SPECIFICATION_NAME`')
          return null
```

File: react/components/BaseSpecificationBadges.tsx [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+        }
+        const slugifiedName = slugify(badge.specification.name)
+        const marginToken = getMarginToken(isVertical, idx === 0, idx === badges.length - 1)
+        return (
+          <div
+            key={`${badge.specification.name}-${valueToShow}`}
+            className={`${styles.badgeContainer} ${styles.badgeContainer}--${slugifiedName} ${marginToken} bg-base flex items-center justify-center`}
+            style={{ borderRadius: '50%', minHeight: "3.5em" }}
```
Comment (lbebber) [inline react/components/BaseSpecificationBadges.tsx]: maybe the default style should be more like, rectangles rather than circles

Comment (lbebber) [reply react/components/BaseSpecificationBadges.tsx]: let me know if you want to clarify

File: react/components/BaseSpecificationBadges.tsx [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+
+        if (!displayValue) {
+          return null
+        }
+        const slugifiedName = slugify(badge.specification.name)
+        const marginToken = getMarginToken(isVertical, idx === 0, idx === badges.length - 1)
+        return (
+          <div
+            key={`${badge.specification.name}-${valueToShow}`}
+            className={`${styles.badgeContainer} ${styles.badgeContainer}--${slugifiedName} ${marginToken} bg-base flex items-center justify-center`}
```
Comment (lbebber) [inline react/components/BaseSpecificationBadges.tsx]: you might already use `vtex.css-handles` here :D

File: react/index.tsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+  blockClass,
+  orientation
+}) => {
+  const { product } = useProduct()
+  return (
+    <BaseSpecificationBadges
+      product={product}
+      blockClass={blockClass}
+      orientation={orientation}
+      groupName={groupName}
```
Comment (lbebber) [inline react/index.tsx]: suggestion, perhaps set it to `specificationGroupName`, to be more consistent with `specificationName` and with the api in general

**Review by lbebber — APPROVED** [review]
These are all suggestions actually, fica a teu criterio

**Pull request merged**